homeassistant/components/risco/alarm_control_panel.py

```diff
@@ -82,28 +109,15 @@ def __init__(self, coordinator, partition_id, code, options):
         for state in self._ha_to_risco:
             self._attr_supported_features |= STATES_TO_SUPPORTED_FEATURES[state]
 
-    def _get_data_from_coordinator(self):
-        self._partition = self.coordinator.data.partitions[self._partition_id]
-
     @property
     def device_info(self) -> DeviceInfo:
```

Comment:
We can set `_attr_device_info`.

Reply:
Consider setting the device info in the child classes instead as it seems these entities represent the main device feature and have different names and ids for the devices.